### PR TITLE
feat(site): Cloudflare video thumbnail fallback for autoplay failure

### PR DESCRIPTION
**Pull request by dante01yoon**
## Summary
- Add `getVideoFrameUrl()` utility that transforms Cloudflare-hosted video URLs into frame extraction URLs (`cdn-cgi/media/mode=frame,time=1s/`)
- Add `poster` attribute to `<video>` elements in `HubWorkflowCard.vue` (grid cards) and `ThumbnailDisplay.astro` (detail pages) for hub API video thumbnails
- Show static fallback `` when video autoplay fails (error/stalled events) — addresses low-power mode, data saver, and restricted browser scenarios
- Only applies to hub API videos (`engcomfy.com` domain); local/curated templates unchanged

Ref: Slack thread on video thumbnail support — Cloudflare thumbnail API task assigned to Jaewon

## Test plan
- [x] Unit tests for `getVideoFrameUrl()` — 6 tests passing
- [ ] Verify poster image loads on grid cards with hub API video thumbnails
- [ ] Verify fallback `` appears when video autoplay is blocked (e.g. throttle network in DevTools)
- [ ] Verify local/curated video templates still autoplay normally (no poster)
- [ ] Verify detail page video controls + poster work correctly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: site/scripts/lib/search/build-index.ts [hunk 1]
```diff
@@ -49,9 +49,18 @@ async function fetchIndexEntries(): Promise<IndexEntry[] | null> {
...
   try {
-    const res = await fetch(`${apiUrl}/api/hub/workflows/index`);
+    // Match hub-api.ts listWorkflowIndex(): preview builds request all statuses
+    // so the search index includes pending/rejected/deprecated workflows too.
+    // Production builds (PUBLIC_APPROVED_ONLY=true) request only approved.
+    const approvedOnly = process.env.PUBLIC_APPROVED_ONLY === 'true';
+    const statuses = approvedOnly
+      ? 'approved'
+      : 'pending,approved,rejected,deprecated';
+    const res = await fetch(`${apiUrl}/api/hub/workflows/index?status=${statuses}`);
```
[the remark this answers is not in the document]
Comment (dante01yoon) [reply site/scripts/lib/search/build-index.ts:59]: Addressed the review feedback — here's what was done:

**Fixed:**
- Extracted `CLOUDFLARE_CDN_DOMAIN` constant in `video-thumbnail.ts`
- Created shared `isVideoFile()`/`isAudioFile()`/`isMediaFile()` helpers in `media-utils.ts`, updated all 3 components
- Added `stalled` event listener + 3s timeout for autoplay suppression detection in `ThumbnailDisplay.astro`
- Removed extra blank lines in `HubWorkflowCard.vue`
- Added edge-case tests (query strings, hash fragments, zero time offset)

**Skipped (intentionally):**
- `fetchIndexEntries()` reuse: runs in Node (`process.env`) vs Astro (`import.meta.env`), direct reuse isn't straightforward
- Refactoring Astro `<script>` to Vue island: larger arch change, non-blocking
- `entries.length > 0 ? entries : null` separation: keeps caller simpler, non-blocking

**Review by christian-byrne — APPROVED** [review]
LGTM
